### DISCUSSION TITLE
cargo build --release --target=x86_64-unknown-linux-musl  (signal: 11, SIGSEGV: invalid memory reference)

### DISCUSSION THREAD

**Discussion by MaksimBy** [qdrant/qdrant#2782] [Q&A]
docker run -v $(pwd):/volume -w /volume -it maksim/musl-rust cargo build --release --target=x86_64-unknown-linux-musl
    Updating git repository `https://github.com/qdrant/tonic`
    Updating crates.io index
    Updating git repository `https://github.com/qdrant/quantization.git`
    Updating git repository `https://github.com/qdrant/wal.git`
  Downloaded actix-macros v0.2.3
  Downloaded mime v0.3.16
  Downloaded encoding_rs_io v0.1.7
  Downloaded miniz_oxide v0.7.1
  Downloaded rand_chacha v0.2.2
  Downloaded parking_lot_core v0.9.5
  Downloaded multimap v0.8.3
  Downloaded memoffset v0.7.1
  Downloaded clap_lex v0.5.0
  Downloaded async-stream-impl v0.3.3
  Downloaded number_prefix v0.4.0
  Downloaded darling_macro v0.20.3
  Downloaded merge_derive v0.1.0
  Downloaded actix-utils v3.0.1
  Downloaded fxhash v0.2.1
  Downloaded colorchoice v1.0.0
  Downloaded alloc-stdlib v0.2.2
  Downloaded cfg-if v1.0.0
  Downloaded validator_types v0.16.0
  Downloaded actix-codec v0.5.0
  Downloaded actix-service v2.0.2
  Downloaded atty v0.2.14
  Downloaded anstyle-query v1.0.0
  Downloaded futures-sink v0.3.28
  Downloaded convert_case v0.4.0
  Downloaded crc v3.0.0
  Downloaded constant_time_eq v0.1.5
  Downloaded ident_case v1.0.1
  Downloaded cow-utils v0.1.2
  Downloaded cexpr v0.6.0
  Downloaded spin v0.5.2
  Downloaded askama_escape v0.10.3
  Downloaded num-derive v0.4.0
  Downloaded bytestring v1.2.0
  Downloaded actix-multipart-derive v0.6.1
  Downloaded atomic_refcell v0.1.11
  Downloaded atomicwrites v0.4.1
  Downloaded getset v0.1.2
  Downloaded rustc-demangle v0.1.21
  Downloaded digest v0.10.6
  Downloaded actix-rt v2.7.0
  Downloaded dlv-list v0.3.0
  Downloaded rand_chacha v0.3.1
  Downloaded block-buffer v0.10.3
  Downloaded time-core v0.1.0
  Downloaded async-stream v0.3.3
  Downloaded matchers v0.1.0
  Downloaded cpufeatures v0.2.5
  Downloaded thread-id v4.0.0
  Downloaded pest_derive v2.5.1
  Downloaded nodrop v0.1.14
  Downloaded fnv v1.0.7
  Downloaded actix-web-codegen v4.2.0
  Downloaded dyn-clone v1.0.10
  Downloaded siphasher v0.3.10
  Downloaded alloc-no-stdlib v2.0.4
  Downloaded which v4.3.0
  Downloaded generic-array v0.14.6
  Downloaded futures-macro v0.3.28
  Downloaded serde-value v0.7.0
  Downloaded futures-io v0.3.28
  Downloaded futures-task v0.3.28
  Downloaded actix-multipart v0.6.1
  Downloaded rmp-serde v1.1.2
  Downloaded tonic-reflection v0.9.2
  Downloaded constant_time_eq v0.3.0
  Downloaded percent-encoding v2.3.0
  Downloaded crc-catalog v2.1.0
  Downloaded crypto-common v0.1.6
  Downloaded encoding-index-singlebyte v1.20141219.5
  Downloaded either v1.8.1
  Downloaded thread_local v1.1.7
  Downloaded actix-web-validator v5.0.1
  Downloaded autocfg v1.1.0
  Downloaded fixedbitset v0.4.2
  Downloaded http-body v0.4.5
  Downloaded version_check v0.9.4
  Downloaded tap v1.0.1
  Downloaded lazy_static v1.4.0
  Downloaded hash32 v0.2.1
  Downloaded hyper-timeout v0.4.1
  Downloaded pkg-config v0.3.26
  Downloaded slice-group-by v0.3.1
  Downloaded validator v0.16.1
  Downloaded pest_generator v2.5.1
  Downloaded json5 v0.4.1
  Downloaded yada v0.5.0
  Downloaded adler v1.0.2
  Downloaded anstyle v1.0.0
  Downloaded glob v0.3.1
  Downloaded arrayvec v0.4.12
  Downloaded merge v0.1.0
  Downloaded sys-info v0.9.1
  Downloaded tower-service v0.3.2
  Downloaded tonic-build v0.10.0
  Downloaded proc-macro-error-attr v1.0.4
  Downloaded tinyvec_macros v0.1.0
  Downloaded anstream v0.5.0
  Downloaded unicode-width v0.1.10
  Downloaded matchit v0.7.0
  Downloaded darling v0.20.3
  Downloaded iana-time-zone v0.1.53
  Downloaded itoa v1.0.5
  Downloaded filetime v0.2.19
  Downloaded ordered-float v3.9.1
  Downloaded blake2-rfc v0.2.18
  Downloaded crossbeam-deque v0.8.2
  Downloaded csv-core v0.1.10
  Downloaded approx v0.5.1
  Downloaded futures-core v0.3.28
  Downloaded pulldown-cmark-to-cmark v10.0.4
  Downloaded impl-more v0.1.6
  Downloaded lindera-dictionary v0.27.0
  Downloaded float_next_after v1.0.0
  Downloaded pin-project-internal v1.0.12
  Downloaded lindera-ko-dic-builder v0.27.0
  Downloaded actix-files v0.6.2
  Downloaded mime_guess v2.0.4
  Downloaded lock_api v0.4.9
  Downloaded ipnet v2.7.0
  Downloaded fs_extra v1.3.0
  Downloaded try-lock v0.2.3
  Downloaded serde_plain v1.0.1
  Downloaded serde_urlencoded v0.7.1
  Downloaded want v0.3.0
  Downloaded byteorder v1.4.3
  Downloaded local-channel v0.1.3
  Downloaded hyper-rustls v0.24.0
  Downloaded anstyle-parse v0.2.0
  Downloaded wyz v0.5.1
  Downloaded tokio-macros v2.1.0
  Downloaded seahash v4.1.0
  Downloaded getrandom v0.2.8
  Downloaded permutation_iterator v0.1.2
  Downloaded miniz_oxide v0.6.2
  Downloaded sct v0.7.0
  Downloaded axum-core v0.3.3
  Downloaded parse-size v1.0.0
  Downloaded futures-channel v0.3.28
  Downloaded actix-cors v0.6.4
  Downloaded ordered-multimap v0.4.3
  Downloaded prost-derive v0.12.0
  Downloaded prost-build v0.11.9
  Downloaded rustc_version v0.4.0
  Downloaded pathdiff v0.2.1
  Downloaded lindera-ipadic-builder v0.27.0
  Downloaded jobserver v0.1.25
  Downloaded lindera-cc-cedict-builder v0.27.0
  Downloaded proc-macro-error v1.0.4
  Downloaded hashring v0.3.2
  Downloaded tokio-io-timeout v1.2.0
  Downloaded crc32fast v1.3.2
  Downloaded colored v2.0.4
  Downloaded bitflags v1.3.2
  Downloaded paste v1.0.11
  Downloaded thiserror v1.0.48
  Downloaded getrandom v0.1.16
  Downloaded shlex v1.1.0
  Downloaded num_cpus v1.16.0
  Downloaded ahash v0.7.6
  Downloaded humantime v2.1.0
  Downloaded io-lifetimes v0.7.5
  Downloaded bincode v1.3.3
  Downloaded quote v1.0.29
  Downloaded time-macros v0.2.6
  Downloaded rmp v0.8.11
  Downloaded tower-layer v0.3.2
  Downloaded ahash v0.8.3
  Downloaded hex v0.4.3
  Downloaded strsim v0.10.0
  Downloaded radium v0.7.0
  Downloaded async-trait v0.1.73
  Downloaded pin-utils v0.1.0
  Downloaded schemars_derive v0.8.13
  Downloaded fastrand v2.0.0
  Downloaded local-waker v0.1.3
  Downloaded peeking_take_while v0.1.2
  Downloaded lindera-decompress v0.27.0
  Downloaded utf8parse v0.2.1
  Downloaded terminal_size v0.1.17
  Downloaded tikv-jemallocator v0.5.4
  Downloaded half v1.8.2
  Downloaded tracing-log v0.1.3
  Downloaded clap_derive v4.4.2
  Downloaded spin v0.9.4
  Downloaded lazycell v1.3.0
  Downloaded raft-proto v0.7.0
  Downloaded if_chain v1.0.2
  Downloaded env_logger v0.10.0
  Downloaded semver v1.0.18
  Downloaded is-terminal v0.4.6
  Downloaded console v0.15.2
  Downloaded tar v0.4.40
  Downloaded futures v0.3.28
  Downloaded rustc-hash v1.1.0
  Downloaded libloading v0.7.4
  Downloaded httparse v1.8.0
  Downloaded num-traits v0.2.16
  Downloaded encoding_index_tests v0.1.4
  Downloaded crossbeam-epoch v0.9.13
  Downloaded form_urlencoded v1.2.0
  Downloaded slog v2.7.0
  Downloaded stable_deref_trait v1.2.0
  Downloaded sync_wrapper v0.1.1
  Downloaded prost-derive v0.11.9
  Downloaded slog-stdlog v4.1.1
  Downloaded clap v4.4.2
  Downloaded ryu v1.0.12
  Downloaded http-range v0.1.5
  Downloaded derive_more v0.99.17
  Downloaded memmap2 v0.7.1
  Downloaded termcolor v1.1.3
  Downloaded yaml-rust v0.4.5
  Downloaded linked-hash-map v0.5.6
  Downloaded toml v0.5.10
  Downloaded validator_derive v0.16.0
  Downloaded scopeguard v1.1.0
  Downloaded socket2 v0.5.3
  Downloaded log v0.4.20
  Downloaded addr2line v0.19.0
  Downloaded serde_derive_internals v0.26.0
  Downloaded language-tags v0.3.2
  Downloaded once_cell v1.18.0
  Downloaded cgroups-rs v0.3.3
  Downloaded ordered-float v2.10.0
  Downloaded lindera-ipadic-neologd-builder v0.27.0
  Downloaded heck v0.4.0
  Downloaded pin-project-lite v0.2.12
  Downloaded memchr v2.5.0
  Downloaded sha1 v0.10.5
  Downloaded serde v1.0.188
  Downloaded funty v2.0.0
  Downloaded io-uring v0.6.1
  Downloaded thiserror-impl v1.0.48
  Downloaded arc-swap v1.6.0
  Downloaded unicode-bidi v0.3.13
  Downloaded unicase v2.6.0
  Downloaded smol_str v0.2.0
  Downloaded actix-server v2.1.1
  Downloaded cookie v0.16.2
  Downloaded actix-router v0.5.1
  Downloaded base64 v0.21.0
  Downloaded rust-ini v0.18.0
  Downloaded serde_cbor v0.11.2
  Downloaded schemars v0.8.13
  Downloaded crossbeam-utils v0.8.14
  Downloaded actix-tls v3.1.0
  Downloaded flate2 v1.0.26
  Downloaded rstar v0.11.0
  Downloaded ppv-lite86 v0.2.17
  Downloaded rand_core v0.5.1
  Downloaded uuid v1.4.1
  Downloaded lindera-core v0.27.0
  Downloaded geo-types v0.7.11
  Downloaded prost v0.12.0
  Downloaded overload v0.1.1
  Downloaded prettyplease v0.2.4
  Downloaded xattr v1.0.1
  Downloaded bitflags v2.3.3
  Downloaded indicatif v0.17.6
  Downloaded docopt v1.1.1
  Downloaded clang-sys v1.4.0
  Downloaded mio v0.8.8
  Downloaded tokio-stream v0.1.11
  Downloaded tokio-util v0.7.4
  Downloaded rayon-core v1.11.0
  Downloaded pest_meta v2.5.1
  Downloaded encoding v0.2.33
  Downloaded lindera-unidic-builder v0.27.0
  Downloaded pin-project v1.0.12
  Downloaded encoding-index-japanese v1.20141219.5
  Downloaded fs4 v0.6.3
  Downloaded whatlang v0.16.2
  Downloaded minimal-lexical v0.2.1
  Downloaded tracing-core v0.1.30
  Downloaded tinyvec v1.6.0
  Downloaded tower v0.4.13
  Downloaded proc-macro2 v1.0.63
  Downloaded prettyplease v0.1.22
  Downloaded futures-executor v0.3.28
  Downloaded libm v0.2.6
  Downloaded indexmap v1.9.2
  Downloaded sharded-slab v0.1.4
  Downloaded untrusted v0.7.1
  Downloaded httpdate v1.0.2
  Downloaded geohash v0.13.0
  Downloaded hashbrown v0.12.3
  Downloaded heapless v0.7.16
  Downloaded slab v0.4.7
  Downloaded time v0.3.17
  Downloaded regex-automata v0.1.10
  Downloaded encoding-index-korean v1.20141219.5
  Downloaded http v0.2.9
  Downloaded signal-hook-registry v1.4.0
  Downloaded sysinfo v0.29.10
  Downloaded irg-kvariants v0.1.0
  Downloaded zstd-safe v6.0.2+zstd.1.5.2
  Downloaded itertools v0.11.0
  Downloaded reqwest v0.11.20
  Downloaded encoding-index-simpchinese v1.20141219.5
  Downloaded rand v0.7.3
  Downloaded futures-util v0.3.28
  Downloaded clap_builder v4.4.2
  Downloaded encoding-index-tradchinese v1.20141219.5
  Downloaded ucd-trie v0.1.5
  Downloaded bytes v1.3.0
  Downloaded rand_core v0.6.4
  Downloaded serde_json v1.0.106
  Downloaded prost-build v0.12.0
  Downloaded tracing-subscriber v0.3.17
  Downloaded rustls-pemfile v1.0.3
  Downloaded anyhow v1.0.75
  Downloaded nu-ansi-term v0.46.0
  Downloaded rustversion v1.0.11
  Downloaded slog-scope v4.4.0
  Downloaded rustls-webpki v0.101.4
  Downloaded typenum v1.16.0
  Downloaded zstd v0.12.2+zstd.1.5.2
  Downloaded hyper v0.14.26
  Downloaded parking_lot v0.12.1
  Downloaded rand_distr v0.4.3
  Downloaded geographiclib-rs v0.2.3
  Downloaded syn v2.0.28
  Downloaded tracing-attributes v0.1.23
  Downloaded webpki-roots v0.25.2
  Downloaded prost-types v0.11.9
  Downloaded idna v0.4.0
  Downloaded darling_core v0.20.3
  Downloaded cc v1.0.78
  Downloaded rustix v0.35.13
  Downloaded rustls v0.21.7
  Downloaded tempfile v3.8.0
  Downloaded rustix v0.36.13
  Downloaded regex-syntax v0.6.28
  Downloaded base64 v0.13.1
  Downloaded rustix v0.38.3
  Downloaded smallvec v1.10.0
  Downloaded serde_qs v0.10.1
  Downloaded prost v0.11.9
  Downloaded regex-syntax v0.7.2
  Downloaded charabia v0.8.3
  Downloaded config v0.13.3
  Downloaded io-lifetimes v1.0.11
  Downloaded serde_derive v1.0.188
  Downloaded tokio-rustls v0.24.1
  Downloaded unicode-ident v1.0.6
  Downloaded backtrace v0.3.67
  Downloaded prost-types v0.12.0
  Downloaded socket2 v0.4.9
  Downloaded ron v0.7.1
  Downloaded pest v2.5.1
  Downloaded bzip2-sys v0.1.11+1.0.8
  Downloaded crossbeam-channel v0.5.6
  Downloaded libc v0.2.147
  Downloaded nom v7.1.1
  Downloaded zstd-sys v2.0.4+zstd.1.5.2
  Downloaded gimli v0.27.0
  Downloaded tokio v1.32.0
  Downloaded url v2.4.1
  Downloaded linux-raw-sys v0.1.4
  Downloaded tikv-jemalloc-sys v0.5.2+5.3.0-patched
  Downloaded linux-raw-sys v0.3.8
  Downloaded tracing v0.1.37
  Downloaded procfs v0.15.1
  Downloaded rand v0.8.5
  Downloaded encoding_rs v0.8.32
  Downloaded portable-atomic v1.3.2
  Downloaded prometheus v0.13.3
  Downloaded libz-sys v1.1.8
  Downloaded pulldown-cmark v0.9.3
  Downloaded itertools v0.10.5
  Downloaded unicode-normalization v0.1.22
  Downloaded axum v0.6.12
  Downloaded actix-http v3.4.0
  Downloaded rocksdb v0.21.0
  Downloaded h2 v0.3.17
  Downloaded rayon v1.7.0
  Downloaded aho-corasick v1.0.2
  Downloaded deunicode v1.3.3
  Downloaded protobuf v2.28.0
  Downloaded raft v0.7.0
  Downloaded brotli-decompressor v2.3.2
  Downloaded petgraph v0.6.2
  Downloaded librocksdb-sys v0.11.0+8.1.1
  Downloaded bindgen v0.65.1
  Downloaded geo v0.26.0
  Downloaded chrono v0.4.30
  Downloaded actix-web v4.4.0
  Downloaded bitvec v1.0.1
  Downloaded nix v0.25.1
  Downloaded regex v1.8.4
  Downloaded object v0.30.0
  Downloaded syn v1.0.107
  Downloaded rustix v0.37.19
  Downloaded robust v1.1.0
  Downloaded earcutr v0.4.2
  Downloaded linux-raw-sys v0.0.46
  Downloaded csv v1.2.2
  Downloaded linux-raw-sys v0.4.3
  Downloaded brotli v3.3.4
  Downloaded fst v0.4.7
  Downloaded ring v0.16.20
  Downloaded protobuf-build v0.14.0
  Downloaded 403 crates (57.5 MB) in 1m 40s (largest was `protobuf-build` at 9.7 MB)
   Compiling zstd-safe v6.0.2+zstd.1.5.2
   Compiling zstd-sys v2.0.4+zstd.1.5.2
   Compiling librocksdb-sys v0.11.0+8.1.1
   Compiling quantization v0.1.0 (https://github.com/qdrant/quantization.git#ff306d0d)
   Compiling tikv-jemalloc-sys v0.5.2+5.3.0-patched
   Compiling mime_guess v2.0.4
   Compiling rust-ini v0.18.0
   Compiling slog-scope v4.4.0
   Compiling clap_derive v4.4.2
   Compiling sharded-slab v0.1.4
   Compiling zstd v0.12.2+zstd.1.5.2
   Compiling tracing-log v0.1.3
   Compiling actix-http v3.4.0
   Compiling serde_plain v1.0.1
   Compiling toml v0.5.10
   Compiling thread_local v1.1.7
   Compiling http-range v0.1.5
   Compiling pathdiff v0.2.1
   Compiling askama_escape v0.10.3
   Compiling tracing-subscriber v0.3.17
   Compiling slog-stdlog v4.1.1
   Compiling tikv-jemallocator v0.5.4
   Compiling sys-info v0.9.1
   Compiling prometheus v0.13.3
   Compiling tonic-reflection v0.9.2
   Compiling config v0.13.3
   Compiling clap v4.4.2
   Compiling colored v2.0.4
   Compiling atty v0.2.14
   Compiling constant_time_eq v0.3.0
   Compiling actix-web v4.4.0
   Compiling serde_qs v0.10.1
   Compiling actix-web-validator v5.0.1
   Compiling actix-files v0.6.2
   Compiling actix-multipart v0.6.1
   Compiling actix-cors v0.6.4
   Compiling rocksdb v0.21.0
   Compiling segment v0.6.0 (/volume/lib/segment)
   Compiling api v1.5.1 (/volume/lib/api)
   Compiling collection v0.4.2 (/volume/lib/collection)
   Compiling storage v0.2.0 (/volume/lib/storage)
   Compiling qdrant v1.5.1 (/volume)
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x3121c13)[0x7f741c305c13]
/lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7f7418e50520]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(_ZN4llvm12DemandedBits15performAnalysisEv+0x12fa)[0x7f741559017a]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(_ZN4llvm8BDCEPass3runERNS_8FunctionERNS_15AnalysisManagerIS1_JEEE+0x73c)[0x7f741558e5fc]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(+0x5f8dea1)[0x7f741558dea1]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(_ZN4llvm11PassManagerINS_8FunctionENS_15AnalysisManagerIS1_JEEEJEE3runERS1_RS3_+0xeab)[0x7f741594bbbb]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(_ZN4llvm27ModuleToFunctionPassAdaptor3runERNS_6ModuleERNS_15AnalysisManagerIS1_JEEE+0x68f)[0x7f741594a665]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(+0x6349fd1)[0x7f7415949fd1]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libLLVM-17-rust-1.73.0-stable.so(_ZN4llvm11PassManagerINS_6ModuleENS_15AnalysisManagerIS1_JEEEJEE3runERS1_RS3_+0xb9)[0x7f7415636587]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x26d723c)[0x7f741b8bb23c]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x26cf1da)[0x7f741b8b31da]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x27c73f6)[0x7f741b9ab3f6]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(_RNvXs1_Cs1HXOGTbQ6wr_18rustc_codegen_llvmNtB5_18LlvmCodegenBackendNtNtNtCs1yZWZnyBLnz_17rustc_codegen_ssa6traits5write19WriteBackendMethods12optimize_fat+0x30)[0x7f741c1a9fa0]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x27c5e7f)[0x7f741b9a9e7f]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x26ca808)[0x7f741b8ae808]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/librustc_driver-453cf35e1dd187fa.so(+0x2659256)[0x7f741b83d256]
/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/libstd-6498d8891e016dca.so(rust_metadata_std_3759e478f3a6c4f2+0x10b295)[0x7f7419155295]
/lib/x86_64-linux-gnu/libc.so.6(+0x94ac3)[0x7f7418ea2ac3]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x44)[0x7f7418f33bf4]
error: could not compile `qdrant` (bin "qdrant")

Caused by:
  process didn't exit successfully: `/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc --crate-name qdrant --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=172 --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto=fat --cfg 'feature="actix-web"' --cfg 'feature="default"' --cfg 'feature="parking_lot"' --cfg 'feature="web"' -C metadata=998d224917b396f5 -C extra-filename=-998d224917b396f5 --out-dir /volume/target/x86_64-unknown-linux-musl/release/deps --target x86_64-unknown-linux-musl -L dependency=/volume/target/x86_64-unknown-linux-musl/release/deps -L dependency=/volume/target/release/deps --extern actix_cors=/volume/target/x86_64-unknown-linux-musl/release/deps/libactix_cors-ae3bf15714cb9f86.rlib --extern actix_files=/volume/target/x86_64-unknown-linux-musl/release/deps/libactix_files-d0b52ee6dceec46d.rlib --extern actix_multipart=/volume/target/x86_64-unknown-linux-musl/release/deps/libactix_multipart-9521bc27902c0d8c.rlib --extern actix_web=/volume/target/x86_64-unknown-linux-musl/release/deps/libactix_web-4f3ed5864557e754.rlib --extern actix_web_validator=/volume/target/x86_64-unknown-linux-musl/release/deps/libactix_web_validator-ba8f850b498ac6a9.rlib --extern anyhow=/volume/target/x86_64-unknown-linux-musl/release/deps/libanyhow-61b661a7c830f69e.rlib --extern api=/volume/target/x86_64-unknown-linux-musl/release/deps/libapi-9db8c88ac5197c36.rlib --extern atty=/volume/target/x86_64-unknown-linux-musl/release/deps/libatty-796908402cd422a0.rlib --extern chrono=/volume/target/x86_64-unknown-linux-musl/release/deps/libchrono-8c46c81a574bf964.rlib --extern clap=/volume/target/x86_64-unknown-linux-musl/release/deps/libclap-8afaf999a216751c.rlib --extern collection=/volume/target/x86_64-unknown-linux-musl/release/deps/libcollection-4f439c4ed015692a.rlib --extern colored=/volume/target/x86_64-unknown-linux-musl/release/deps/libcolored-9cda9aad9f2e1024.rlib --extern common=/volume/target/x86_64-unknown-linux-musl/release/deps/libcommon-1c02a66b21c79ba9.rlib --extern config=/volume/target/x86_64-unknown-linux-musl/release/deps/libconfig-9f90eb1b59d8e72d.rlib --extern constant_time_eq=/volume/target/x86_64-unknown-linux-musl/release/deps/libconstant_time_eq-7b9c216f5422f755.rlib --extern futures=/volume/target/x86_64-unknown-linux-musl/release/deps/libfutures-85cdf0b6137ddd01.rlib --extern futures_util=/volume/target/x86_64-unknown-linux-musl/release/deps/libfutures_util-f63565be83ef8f84.rlib --extern itertools=/volume/target/x86_64-unknown-linux-musl/release/deps/libitertools-1b5854dcbcc0c93f.rlib --extern log=/volume/target/x86_64-unknown-linux-musl/release/deps/liblog-0d5373215fdbdd41.rlib --extern num_traits=/volume/target/x86_64-unknown-linux-musl/release/deps/libnum_traits-c1bd0188377d5b45.rlib --extern num_cpus=/volume/target/x86_64-unknown-linux-musl/release/deps/libnum_cpus-14cdfa42cdf21e2c.rlib --extern parking_lot=/volume/target/x86_64-unknown-linux-musl/release/deps/libparking_lot-7c108e4ff8669953.rlib --extern prometheus=/volume/target/x86_64-unknown-linux-musl/release/deps/libprometheus-f4fb8cb18d58a560.rlib --extern prost=/volume/target/x86_64-unknown-linux-musl/release/deps/libprost-92a85f5e8dec26d6.rlib --extern raft=/volume/target/x86_64-unknown-linux-musl/release/deps/libraft-3a09c102af369797.rlib --extern raft_proto=/volume/target/x86_64-unknown-linux-musl/release/deps/libraft_proto-a105cd5a507e0d68.rlib --extern reqwest=/volume/target/x86_64-unknown-linux-musl/release/deps/libreqwest-a60b5176e3509772.rlib --extern rustls=/volume/target/x86_64-unknown-linux-musl/release/deps/librustls-a3ff50f2a65f13b2.rlib --extern rustls_pemfile=/volume/target/x86_64-unknown-linux-musl/release/deps/librustls_pemfile-341c442a4c989c9f.rlib --extern schemars=/volume/target/x86_64-unknown-linux-musl/release/deps/libschemars-87404b124e40815f.rlib --extern segment=/volume/target/x86_64-unknown-linux-musl/release/deps/libsegment-ea34d65281503b8a.rlib --extern serde=/volume/target/x86_64-unknown-linux-musl/release/deps/libserde-16439c36ec16e250.rlib --extern serde_cbor=/volume/target/x86_64-unknown-linux-musl/release/deps/libserde_cbor-8d7835f884ec7fdb.rlib --extern serde_json=/volume/target/x86_64-unknown-linux-musl/release/deps/libserde_json-202e27170d37e1e4.rlib --extern slog=/volume/target/x86_64-unknown-linux-musl/release/deps/libslog-08530459e66ea5ce.rlib --extern slog_stdlog=/volume/target/x86_64-unknown-linux-musl/release/deps/libslog_stdlog-dee5487b5802a181.rlib --extern storage=/volume/target/x86_64-unknown-linux-musl/release/deps/libstorage-ba32b3243263cdf3.rlib --extern sys_info=/volume/target/x86_64-unknown-linux-musl/release/deps/libsys_info-eb27faff54c3c38c.rlib --extern tar=/volume/target/x86_64-unknown-linux-musl/release/deps/libtar-86b8acd0f9e6d06f.rlib --extern thiserror=/volume/target/x86_64-unknown-linux-musl/release/deps/libthiserror-02f012d2e36055eb.rlib --extern tikv_jemallocator=/volume/target/x86_64-unknown-linux-musl/release/deps/libtikv_jemallocator-34b80739ff273acf.rlib --extern tokio=/volume/target/x86_64-unknown-linux-musl/release/deps/libtokio-3a50dd986827ac4f.rlib --extern tonic=/volume/target/x86_64-unknown-linux-musl/release/deps/libtonic-2dfda7bf68f61124.rlib --extern tonic_reflection=/volume/target/x86_64-unknown-linux-musl/release/deps/libtonic_reflection-874f44a1959dc05b.rlib --extern tower=/volume/target/x86_64-unknown-linux-musl/release/deps/libtower-114818da9ea16c17.rlib --extern tower_layer=/volume/target/x86_64-unknown-linux-musl/release/deps/libtower_layer-1406fa24d0ef09dc.rlib --extern tracing=/volume/target/x86_64-unknown-linux-musl/release/deps/libtracing-032a6dce1c100bfd.rlib --extern tracing_log=/volume/target/x86_64-unknown-linux-musl/release/deps/libtracing_log-25690b1dab7fe7a1.rlib --extern tracing_subscriber=/volume/target/x86_64-unknown-linux-musl/release/deps/libtracing_subscriber-8b076f8f0be8136e.rlib --extern uuid=/volume/target/x86_64-unknown-linux-musl/release/deps/libuuid-2712ab62ad298df3.rlib --extern validator=/volume/target/x86_64-unknown-linux-musl/release/deps/libvalidator-ff32e50933fa5c75.rlib --extern wal=/volume/target/x86_64-unknown-linux-musl/release/deps/libwal-7ccdcb0721c108e2.rlib -L native=/volume/target/x86_64-unknown-linux-musl/release/build/ring-57b99276003d32ee/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/zstd-sys-a003e63fbbfdd11d/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/quantization-1333e55662974a98/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/librocksdb-sys-016e3d5b0c3cb9c5/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/librocksdb-sys-016e3d5b0c3cb9c5/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/sys-info-23c785be8a0b336d/out -L native=/volume/target/x86_64-unknown-linux-musl/release/build/tikv-jemalloc-sys-ea5cb0769025c7f4/out/build/lib` (signal: 11, SIGSEGV: invalid memory reference)

**Comment by timvisee** [comment]
Hi @MaksimBy. Building Qdrant for musl is not supported at this time. This is due to limited or missing support in some of the dependencies we use.

We may pick this up eventually though as part of releasing statically linked binaries: https://github.com/qdrant/qdrant/issues/2656
